### PR TITLE
[CBRD-24463] Enable communication histogram

### PR DESCRIPTION
**Pull request by hgryoo**
http://jira.cubrid.org/browse/CBRD-24463

**Motivation**
The histogram is a good tool to measure the performance of each communication function when you simply executing a query.

**Implementation**
- improvement: histogram functionality is moved and rewritten in *network_histogram* files. APIs are defined to support cs/sa behavior in general routine
- improvement: histogram functionlity is enabled in network_cl.c: `#defiend (HISTO)` is removed.
- refactor: the `net_request`'s name lists are duplicated on client/server. the duplicated lists are moved to *network_common.cpp*
- refactor: split net_request definition to a new file, *network_request_def.hpp*
- refactor: remove histogram-related code in *network_sr.c* (they are already disabled and the histogram is not used in the server module)
- refactor: remove histogram functionality in `SQL_LOG2` of CAS.

**Remarks**
N/A

### CODE REVIEW

**Review by hyunikn — APPROVED** [review]

File: src/communication/network_histogram.hpp [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+/*
+ * network_histogram..hpp - Add instrumentations to the client side to get histogram of network requests
```
Comment (youngjinj) [inline src/communication/network_histogram.hpp]: There are 2 dots, is it a typo?
```suggestion
 * network_histogram.hpp - Add instrumentations to the client side to get histogram of network requests
```

**Review by beyondykk9 — APPROVED** [review]

File: src/communication/network_request_def.hpp [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  int action_attribute; /* bitmask, see net_req_act */
+  net_server_func processing_function;
+
+  /* use get_net_request_name () */
+  // const char *name;
+
+  net_request () = default;
+};
+
+#endif /* _NETWORK_REQUEST_DEF_HPP_ */
```
Comment (youngjinj) [inline src/communication/network_request_def.hpp]: No newline at end of file.

**Review by youngjinj — APPROVED** [review]

File: src/communication/network_histogram.hpp (hunk 1 above)
Comment (hgryoo) [reply src/communication/network_histogram.hpp]: thank you

File: src/communication/network_request_def.hpp (hunk 2 above)
Comment (hgryoo) [reply src/communication/network_request_def.hpp]: fixed, thank you

**Review by shparkcubrid — APPROVED** [review]

File: src/communication/network_histogram.cpp [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+#endif /* !CS_MODE */
+}
+
+void
+histo_finish_request (int request, int received)
+{
+#if defined (CS_MODE)
+  net_histo_context.finish_request (request, received);
+#endif /* !CS_MODE */
+}
```
Comment (ctshim) [inline src/communication/network_histogram.cpp]: Enter, please.

**Review by ctshim — APPROVED** [review]

File: src/communication/network_histogram.cpp (hunk 3 above)
Comment (hgryoo) [inline src/communication/network_histogram.cpp]:
```suggestion
}

```

**Pull request merged**